tests/v2/test_utils_keyfile.nim

```diff
@@ -0,0 +1,120 @@
+{.used.}
+
+import
+  std/[json, os],
+  testutils/unittests, chronos, chronicles,
+  eth/keys
+import
+  ../../waku/v2/utils/keyfile,
+  ../test_helpers
+
+from ../../waku/v2/protocol/waku_noise/noise_utils import randomSeqByte
+
+suite "KeyFile test suite":
+
+  let rng = newRng()
+
+  test "Create/Save/Load single keyfile":
+
+    let password = "randompassword"
+
+    let filepath = "./test.keyfile"
+    defer: removeFile(filepath)
+
+    var secret = randomSeqByte(rng[], 300)
+    let keyfile = createKeyFileJson(secret, password)
+
+    check:
+      keyfile.isOk()
+      saveKeyFile(filepath, keyfile.get()).isOk()
+
+    var decodedSecret = loadKeyFile("test.keyfile", password)
+
+    check:
+      decodedSecret.isOk()
+      secret == decodedSecret.get()
+
+  test "Create/Save/Load multiple keyfiles in same file":
+
+    let password1 = "password1"
```

Comment:
May we also have a test for an empty password? it should work in general, but as it is a corner case, it may result in some abrupt behavior.